src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -67,79 +73,12 @@ const PendingLink: React.FC<Pick<Props, 'pending'>> = ({ pending }) => {
   )
 }
 
-interface OrderDetailsProps extends Pick<Props, 'buyToken' | 'sellToken' | 'pending'> {
-  price: string
-}
-
-const OrderDetails: React.FC<OrderDetailsProps> = ({ price, buyToken, sellToken, pending }) => (
-  <div className="container order-details">
-    <div>Sell</div>
-    <div>
-      <Highlight color={pending ? 'grey' : ''}>1</Highlight>
-    </div>
-    <div>
-      <strong>{safeTokenName(sellToken)}</strong>
-    </div>
-
-    <div>
-      for <strong>at least</strong>
-    </div>
-    <div>
-      <Highlight color={pending ? 'grey' : 'red'}>{price}</Highlight>
-    </div>
-    <div>
-      <strong>{safeTokenName(buyToken)}</strong>
-    </div>
-  </div>
-)
-
-interface UnfilledAmountProps extends Pick<Props, 'sellToken' | 'pending'> {
-  unfilledAmount: string
-  unlimited: boolean
-}
-
-const UnfilledAmount: React.FC<UnfilledAmountProps> = ({ sellToken, unfilledAmount, unlimited, pending }) => (
-  <div className={'container' + (unlimited ? '' : ' sub-columns two-columns')}>
-    {unlimited ? (
-      <Highlight color={pending ? 'grey' : ''}>no limit</Highlight>
-    ) : (
-      <>
-        <div>{unfilledAmount}</div>
-        <div>
-          <strong>{safeTokenName(sellToken)}</strong>
-        </div>
-      </>
-    )}
-  </div>
-)
-
-interface AccountBalanceProps extends Pick<Props, 'sellToken' | 'isOverBalance'> {
-  accountBalance: string
-}
-
-const AccountBalance: React.FC<AccountBalanceProps> = ({ sellToken, accountBalance, isOverBalance }) => (
-  <div className="container sub-columns three-columns">
-    <div>{accountBalance}</div>
-    <strong>{safeTokenName(sellToken)}</strong>
-    <div className="warning">{isOverBalance && <FontAwesomeIcon icon={faExclamationTriangle} />}</div>
-  </div>
-)
-
-interface ExpiresProps extends Pick<Props, 'pending'> {
-  isNeverExpires: boolean
-  expiresOn: string
-}
-
-const Expires: React.FC<ExpiresProps> = ({ pending, isNeverExpires, expiresOn }) => (
-  <div>{isNeverExpires ? <Highlight color={pending ? 'grey' : ''}>Never</Highlight> : expiresOn}</div>
-)
-
-interface Props {
-  sellToken: TokenDetails
-  buyToken: TokenDetails
-  order: AuctionElement
-  isOverBalance: boolean
-  pending?: boolean
+function displayTokenSymbolOrLink(token: TokenDetails): React.ReactNode | string {
+  const displayName = safeTokenName(token)
+  if (displayName.startsWith('0x')) {
```

Comment:
If it's a check for address, then a better check would be a regex or `isAddress` from `web3-utils`

Reply:
`safeTokenName` returns the already abbreviated address (`0xXXXX...XXXX`) when no `name` nor `symbol` are present.
`isAddress` wouldn't work for this.